### PR TITLE
ngx_http_realip_module takes X-User-IP to get the real client IP address

### PR DESCRIPTION
**Pull request by mangchiandjjoe**

### CODE REVIEW

File: start_esp/nginx-auto.conf.template [hunk 1]
```diff
@@ -48,10 +48,7 @@ http {
 % endif
 % endfor
 
-  set_real_ip_from  0.0.0.0/0;
-  set_real_ip_from  0::/0;
-  real_ip_header    X-Forwarded-For;
-  real_ip_recursive on;
```
Comment (qiwzhang) [inline start_esp/nginx-auto.conf.template]: can we have both?  For GCP, we can use X-User-IP,  for others,  we may have to use XFF

Comment (mangchiandjjoe) [reply start_esp/nginx-auto.conf.template]: ngx_http_realip_module doesn't allow duplication. 
https://github.com/nginx/nginx/blob/master/src/http/modules/ngx_http_realip_module.c#L425

And it doesn't support multiple header in the setting either.
https://github.com/nginx/nginx/blob/master/src/http/modules/ngx_http_realip_module.c#L214

We can add an command line argument to choose which header.

Comment (qiwzhang) [reply start_esp/nginx-auto.conf.template]: ideally, we should support both,  check x-user-ip first, then xff.   users should not need to config it.

Comment (mangchiandjjoe) [reply start_esp/nginx-auto.conf.template]: To support both, we need to either implement similar logic in esp or modify existing module.

Comment (qiwzhang) [reply start_esp/nginx-auto.conf.template]: let us go with implementing it ourrself.  it has two benefits
1) not need to change nginx config
2) it is more flexible, we can do any ways we wanted.

Comment (mangchiandjjoe) [reply start_esp/nginx-auto.conf.template]: If we implement this feature in esp, are we going to turn on this override by default or create a option to turn it on?

Comment (qiwzhang) [reply start_esp/nginx-auto.conf.template]: I suggest on by default.

**Comment by mangchiandjjoe** [comment]
Closing current PR for another PR

**Pull request closed without merging**